ADAL/src/validation/ADAuthorityValidation.m

```diff
@@ -249,13 +252,15 @@ - (void)validateAADAuthority:(NSURL *)authority
     });
 }
 
-- (void)requestAADValidation:(NSURL *)authority
+- (void)requestAADValidation:(NSURL *)authorityUrl
                requestParams:(ADRequestParameters *)requestParams
              completionBlock:(ADAuthorityValidationCallback)completionBlock
 {
+    __auto_type authority = [[MSIDAADAuthority alloc] initWithURL:authorityUrl context:nil error:nil];
```

Comment:
what's the benefit of using MSIDAADAuthority here?

Reply:
We just explicitly show that we expect only AAD authority here. Also added logic that returns right away if parsing to MSIDAADAuthority failed.